### ISSUE TITLE
Changing SQL computed column - migration generates wrong SQL; ALTER instead of DROP/ADD

### ISSUE THREAD

**Issue by skorunka** [dotnet/efcore#7085]
### Steps to reproduce
Im using, SQL Server 2008, "Microsoft.EntityFrameworkCore": "1.0.1",
Code:
```C#
		protected override void OnModelCreating(ModelBuilder modelBuilder)
		{
			modelBuilder.Entity<Test>()
						.Property(p => p.IdFormatted)
						.HasComputedColumnSql("9000000 + Id");
```
Add-Migration FormattedId1
```C#
		protected override void OnModelCreating(ModelBuilder modelBuilder)
		{
			modelBuilder.Entity<Test>()
						.Property(p => p.IdFormatted)
						.HasComputedColumnSql("(CONVERT(CHAR(7), 9000000 + Id))");
```
```
Add-Migration FormattedId1
Update-Database:
System.Data.SqlClient.SqlException (0x80131904): Incorrect syntax near the keyword 'AS'. 
    at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction) 
    at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose) 
    at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady) 
    at System.Data.SqlClient.SqlCommand.RunExecuteNonQueryTds(String methodName, Boolean async, Int32 timeout, Boolean asyncWrite) 
    at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, String methodName, Boolean sendToPipe, Int32 timeout, Boolean asyncWrite) 
    at System.Data.SqlClient.SqlCommand.ExecuteNonQuery() 
    at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.Execute(IRelationalConnection connection, String executeMethod, IReadOnlyDictionary`2 parameterValues, Boolean openConnection, Boolean closeConnection) 
    at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteNonQuery(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues, Boolean manageConnection) 
    at Microsoft.EntityFrameworkCore.Migrations.Internal.MigrationCommandExecutor.ExecuteNonQuery(IEnumerable`1 migrationCommands, IRelationalConnection connection) 
    at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.Migrate(String targetMigration) 
    at Microsoft.EntityFrameworkCore.Design.MigrationsOperations.UpdateDatabase(String targetMigration, String contextType) 
    at Microsoft.EntityFrameworkCore.Tools.Cli.DatabaseUpdateCommand.<>c__DisplayClass0_0.<Configure>b__0() 
    at Microsoft.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args) 
    at Microsoft.EntityFrameworkCore.Tools.Cli.Program.Main(String[] args) 
```
### The issue

Wrong SQL is generated by migrations to change computed colum on SQL server.

Generated SQL:
```sql
DECLARE @var0 sysname;
SELECT @var0 = [d].[name]
FROM [sys].[default_constraints] [d]
INNER JOIN [sys].[columns] [c] ON [d].[parent_column_id] = [c].[column_id] AND [d].[parent_object_id] = [c].[object_id]
WHERE ([d].[parent_object_id] = OBJECT_ID(N'TLEDOL.Tests') AND [c].[name] = N'IdFormatted');
IF @var0 IS NOT NULL EXEC(N'ALTER TABLE [TLEDOL].[Tests] DROP CONSTRAINT [' + @var0 + ']');
ALTER TABLE [TLEDOL].[Tests] ALTER COLUMN [IdFormatted] AS (CONVERT(CHAR(7), 9000000 + Id));
```
Should be:
```sql
ALTER TABLE [TLEDOL].[Tests] DROP COLUMN [IdFormatted];
ALTER TABLE [TLEDOL].[Tests] ADD [IdFormatted] AS (CONVERT(CHAR(7), 9000000 + Id));
```
### Further technical details

http://stackoverflow.com/questions/2372655/how-can-i-alter-this-computed-column-in-sql-server-2008
http://www.sqlservercentral.com/Forums/Topic995032-338-1.aspx#bm995387

**Comment by rowanmiller** [comment]
The current workaround would be to manually edit the scaffolded migration to do the drop and create.

**Comment by bricelam** [comment]
This sounds familiar. It may have been fixed in 1.1.0.

**Comment by skorunka** [comment]
@bricelam Yes, 1.1.0 works fine.
